### PR TITLE
Add JSDocs

### LINKED ISSUE #448: suggestion: complete documentation
I suggest having documentation for every symbol within this project. One can check for remaining symbols missing documentation by running `deno doc --lint mod.ts`.

### PR DESCRIPTION
**Pull request by bombillazo**
Fixes #448

- Add JSDocs
- Export more types and classes

### CODE REVIEW

File: query/transaction.ts [hunk 1]
```diff
@@ -386,9 +418,34 @@ export class Transaction {
     query: string,
     args?: QueryArguments,
   ): Promise<QueryArrayResult<T>>;
+  /**
+   * Use the configuration object for more advance options to execute the query
+   *
+   * ```ts
+   * import { Client } from "./client.ts";
```
Comment (iuioiua) [inline query/transaction.ts:425]:
```suggestion
   * import { Client } from "https://deno.land/x/postgres/mod.ts";
```
Ditto for all other snippets. This should fix your errors.

Comment (bombillazo) [reply query/transaction.ts:425]: even when it is in a comment, the linter thinks it is a real path? That was just an example code

Comment (iuioiua) [reply query/transaction.ts:425]: Yep 🙂

**Comment by iuioiua** [comment]
Why remove the full remote imports? They'd be helpful for users wanting to copy/paste snippets.

**Comment by bombillazo** [comment]
Yeah, I saw it wasn't the right approach; I fixed it now.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**